source/NVDAObjects/UIA/__init__.py

```diff
@@ -824,6 +825,16 @@ def _get_shouldAllowUIAFocusEvent(self):
 		except COMError:
 			return True
 
+	_lastLiveRegionChangeInfo=(None,None) #: Keeps track of the last live region change (text, time)
+	def _get__shouldAllowUIALiveRegionChangeEvent(self):
+		oldText,oldTime=self._lastLiveRegionChangeInfo
+		newText=self.name
+		newTime=time.time()
+		if newText==oldText and oldTime is not None and (newTime-oldTime)<0.5:
+			return False

```

Comment:
Do we still want to update the cached time if the text is equal? Unless I'm missing something, this will still spam every half second if the same text keeps firing.